modules/dnn/test/test_halide_layers.cpp

```diff
@@ -583,7 +583,7 @@ TEST_P(NoParamActivation, Accuracy)
     testInPlaceActivation(lp, backendId, targetId);
 }
 INSTANTIATE_TEST_CASE_P(Layer_Test_Halide, NoParamActivation, Combine(
-/*type*/ Values("TanH", "Sigmoid", "AbsVal", "BNLL"),
+/*type*/ Values("TanH", "Sigmoid", "AbsVal", "BNLL", "Swish", "Mish"),
```

Comment:
Good choice! 😄

Reply:
The CUDA tests are passing for the Swish activation.

The test for `DNN_TARGET_CUDA` is passing for the Mish activation. The test for `DNN_TARGET_CUDA_FP16` is failing due.

```
[ RUN      ] Layer_Test_Halide/NoParamActivation.Accuracy/21, where GetParam() = ("Mish", CUDA/CUDA_FP16)
.../opencv/modules/dnn/test/test_common.impl.hpp:68: Failure
Expected: (normL1) <= (l1), actual: 0.0288897 vs 0.004
.../opencv/modules/dnn/test/test_common.impl.hpp:71: Failure
Expected: (normInf) <= (lInf), actual: 0.0909604 vs 0.02
```

You will have to increase the error tolerance for Mish activation when target is `DNN_TARGET_CUDA_FP16`. I can't think of an easy and correct way to do this. In the worst case, the test can be skipped for the FP16 target.

Reply:
The default Thresholds are being called from here https://github.com/thebhatman/opencv/blob/f6221044661dedbf4ad729c8f24c2b72ff37aca0/modules/dnn/test/test_common.hpp#L113. I don't see a way to pass the thresholds as arguments without changing the prototype of `testInPlaceActivation` .

Reply:
I think you should add new parameters (`l1` and `lInf`) to `test` and `testInPlaceActivation`. Their default values would be `0.0`.

Inside the `test` function:
- if `l1` and `lInf` arguments are zero, set them to the values given by `getDefaultThresholds`
- if `l1` and `lInf arguments are not zero, use them

@dkurt is this  ok?

Reply:
@YashasSamaga, yes, that would be fine. Thanks!